pkg/machine/apple/apple.go

```diff
@@ -225,6 +227,19 @@ func StartGenericAppleVM(mc *vmconfigs.MachineConfig, cmdBinary string, bootload
 		cmd.Args = append(cmd.Args, "--gui") // add command line switch to pop the gui open
 	}
 
+	if mc.LibKrunHypervisor != nil {
+		// use sysctl to get the CPUID information
+		cpuBrandString, err := unix.Sysctl("machdep.cpu.brand_string")
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// only M3 processors and newer can support nested virtualization
+		if strings.Contains(cpuBrandString, "M3") || strings.Contains(cpuBrandString, "M4") {
```

Comment:
This looks like constant maintenance cost, if there is an M5 chip it would be fair to assume it would work there as well I think. And I rather not have that updated every year because we forget until someone reports a bug.

(And in general trying to match the logic from libkrun here again just introduces the possibility of drift between the two implementations. I rather have some smart feature flag in krunkit that enabled the virt but only if possible? Any reason this is not the default in libkrun?)

Reply:
So far we tried to avoid adding policy in libkrun, leaving that to the caller. What we can do is add a new API to allow callers to check if nested is supported in the platform. WDYT @jakecorrenti ?

Reply:
Now that we've bumped krunkit to 0.2.1, it includes the `krun_check_nested_virt` API. This now allows Podman to do `--nested`, but if the host isn't supported it will ignore the argument and continue starting the VM as normal.